WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -392,12 +392,30 @@ def time_until_competition
     }
   end
 
-  def currency_convert
-    update_rates_if_needed
-    converted = Money.new(params[:value], params[:from_currency]).exchange_to(params[:to_currency])
+  def calculate_dues
+    country_iso2 = Country.find_by(id: params[:country_id])&.iso2
+    country_band = CountryBand.find_by(iso2: country_iso2)&.number
+
+    update_exchange_rates_if_needed
+    input_money_us_dollars = Money.new(params[:entry_fee_cents].to_i, params[:currency_code]).exchange_to("USD").amount
+
+    registration_fee_dues_us_dollars = input_money_us_dollars * CountryBand::PERCENT_REGISTRATION_FEE_USED_FOR_DUE_AMOUNT
+    country_band_dues_us_dollars = CountryBand::BANDS[country_band][:value] if country_band.present? && country_band > 0
+
+    # times 100 because later currency conversions require lowest currency subunit, which is cents for USD
+    price_per_competitor_us_cents = [registration_fee_dues_us_dollars, country_band_dues_us_dollars].compact.max * 100
+    estimated_dues = "" # return empty string if no competitor limit
+
+    if params[:competitor_limit_enabled]
+      estimated_dues_us_cents = price_per_competitor_us_cents * params[:competitor_limit].to_i
+      estimated_dues = Money.new(estimated_dues_us_cents, "USD").exchange_to(params[:currency_code]).format
+    end
```

Comment:
Suggestion to avoid empty variables:
```ruby
if params[:competitor_limit_enabled]
  estimated_dues_us_cents = foo
  estimated_dues = bar
  
  return render json: {
    stuff
  } 
else
  price_per_competitor = foo
  
  return render json: {
    stuff
  }
end
```